src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.netcoreapp.cs

```diff
@@ -84,5 +84,52 @@ public void Dictionary_Generic_RemoveKey_DefaultKeyContainedInDictionary(int cou
         }
 
         #endregion
+
+        #region EnsureCapacity
+        
+        [Fact]
+        public void EnsureCapacity_NegativeCapacityRequested_Throws()
+        {
+            Dictionary<string, string> dictionary = new Dictionary<string, string>();
+            Assert.Throws<ArgumentOutOfRangeException>(() => dictionary.EnsureCapacity(-1));
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(3)]
+        public void EnsureCapacity_DefaultCapacityOnEmptyDictionary_ReturnsCapacityRequestedLargerOrEqual(int requestedCapacity)
+        {
+            Dictionary<string, string> dictionary = new Dictionary<string, string>();
+            Assert.True(requestedCapacity <= dictionary.EnsureCapacity(requestedCapacity));
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(3)]
+        public void EnsureCapacity_CapacityRequestedSmallerThanCurrentCapacity_CapacityUnchanged(int requestedCapacity)
+        {
+            Dictionary<string, string> dictionary = new Dictionary<string, string>();
+
+            var capacity = dictionary.EnsureCapacity(requestedCapacity);
+
+            Assert.Equal(capacity, dictionary.EnsureCapacity(requestedCapacity-1));
+        }
+        
+        [Theory]
+        [InlineData(1)]
+        [InlineData(5)]
+        public void EnsureCapacity_CapacityRequestedSmallerThanCount_SetsCapacityToLargerOrEqualToExistingCount(int count)
+        {
+            Dictionary<int, int> dictionary = new Dictionary<int, int>();
+            for (int i = 0; i < count; i++)
+            {
+                dictionary.Add(i, i);
+            }
+
+            Assert.True(dictionary.Count <= dictionary.EnsureCapacity(count-1));
+        }
+
+        #endregion
```

Comment:
You don't have a test for the 90% case, which is increasing the capacity of a non empty dictionary.

In that test you could add a bunch of items, call EnsureCapacity(0) to get the current capacity, call EnsureCapacity(X) where X is equal the current capacity and check you get the same value back, then call EnsureCapacity(Y) where Y is larger and check you get the larger one back. Then EnsureCapacity(0) and check it's Y.